### DISCUSSION TITLE
crop ct scan deep from the top

### DISCUSSION THREAD

**Discussion by oscarramirezs** [Project-MONAI/MONAI#1929] [General]
HELLO, 
I have a question, can someone help me? sorry it may be a silly question

I have some Keen CT scan, each image is 512x512  and deep changes from one CT scan to another (ex; sometime 120 images to 900 images), sometimes the CT start from the ankle and some time from near the knee,

so when I use crop = SpatialCropd(keys=["image", "label"], roi_start=[0, 0, 0], roi_end=[512, 512, 256]), the scanner is cut from the bottom to the top, but I want to start cropping from the top to the Botton because always the keen is in the top. 

How can I cut the images starting always from the last one, ? ( the last one change for each ct) 

thank you for your help

Oscar

**Comment by function2-llx** [comment]
I think [`Orientation`](https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/spatial/array.py#L205) is what you want, which reorients images to a fixed orientation. Check the "Reorientation to a designated axes codes" section in [this tutorial](https://github.com/Project-MONAI/tutorials/blob/master/modules/3d_image_transforms.ipynb).

**Reply by oscarramirezs** [reply]
Hi, Function2
thanks, but it doesn't work,

I have some CTs with different depth sizes for example (512 * 512 * 482) another (512 * 512 * 658) another (512 * 512 * 800),
I would like to have a final size of 512 * 512 * 256 for all CTs , but I'm interested in the last images and not the first ones, so when I use SpatialCropd (keys = ["image", "label"], roi_start = [], roi_end = [,]), I get the first 256 images in each CT, and I want to get are the 256 latest images

thank you,
I don't know if is clear

Oscar

**Reply by function2-llx** [reply]
Seems that I misunderstood your question at first. 

TL;DR: use `Lambda(lambda x: x[:, :, -256])`.

In fact something like `roi_start=(0, 0, -256)` and `roi_end=(None, None, None)` should work, which makes sense like slicing an array, however this currently does not work in MONAI (as I've mentioned in #1935). But some workaround like `Lambda(lambda x: x[:, :, -256])` will work perfectly.

**Reply by oscarramirezs** [reply]
Hi function2-llx 

Thank your help, 

sorry I'm  new to python, where should I use lambda function? I have my transformation function as shown below, if you can help me, I will appreciate it very much.

`train_transforms` = Compose(
    [
        LoadImaged(keys=["image", "label"]),
        AddChanneld(keys=["image", "label"]),
        **lambd = Lambda(func=lambda x: x[:, :, -256]),**
        SpatialCropd(keys=["image", "label"],roi_start=[0, 0, -256], roi_end=[None, None, None]),     
        RandFlipd(keys=["image", "label"], prob=0.33, spatial_axis=0).set_random_state(rseed),
        RandGaussianNoised(keys=["image"], prob=0.1, mean=48, std=330).set_random_state(rseed),        
        ToTensord(keys=["image", "label"]),
    ]
)`

thank you 
Oscar

**Reply by function2-llx** [reply]
@oscarramirezs Check if this works:
```python
train_transforms = Compose([
    LoadImaged(keys=["image", "label"]),
    AddChanneld(keys=["image", "label"]),
    Lambdad(keys=["image", "label"], func=lambda x: x[:, :, :, -256]),
    RandFlipd(keys=["image", "label"], prob=0.33, spatial_axis=0).set_random_state(rseed),
    RandGaussianNoised(keys=["image"], prob=0.1, mean=48, std=330).set_random_state(rseed),
    ToTensord(keys=["image", "label"]),
])
```

**Reply by oscarramirezs** [reply]
Hi, @function2-llx 

it works thanks 

I changed it a little for

Lambdad(keys=["image", "label"], func=lambda x: x[ :, :, :, -256:]), 




Thanks a lot @function2-llx 

Oscar

**Reply by function2-llx** [reply]
Congrats!

Yeah a typo missing `func` key, I've already edited it.

**Comment by function2-llx** [comment]
As the discussion above, we could achieve it with `Lambda`. However I think it will be cool if I could do the same thing just with `SpatialCrop`.

**Comment by rijobro** [comment]
Hi, suggested PR [here](https://github.com/Project-MONAI/MONAI/pull/1954). If you wanted to leave the last 100 in the z-direction and not touch anything else, this would be done with:

```python
SpatialCrop(roi_slices=[slice(None), slice(None), slice(-100, None)])
```

**Reply by rijobro** [reply]
This has the advantage of being invertible.

**Reply by function2-llx** [reply]
Hi, may I ask that if `roi_slices` is supported, is it necessary to keep `roi_start` and `roi_end`?

**Reply by rijobro** [reply]
You can use `roi_slices` **or** `roi_start` and `roi_end`. By the way, the PR has been merged, so you should be able to use it if you're using an up-to-date version of MONAI master.

**Reply by rijobro** [reply]
I actually just checked the code. If `roi_slices` is provided, `roi_start` and `roi_end` won't be used if they have been provided, but an error won't be thrown either. They will just be ignored.

**Reply by Nic-Ma** [reply]
Hi @rijobro , I submitted a bug related to this question, could you please help take a look? It's urgent issue for our release: https://github.com/Project-MONAI/MONAI/issues/2087.

Thanks in advance.
